src/operator/controllers/intents_reconcilers/networkpolicy/reconciler.go

```diff
@@ -279,7 +295,8 @@ func (r *Reconciler) setNetworkPolicyOwnerReferenceIfNeeded(ctx context.Context,
 	return errors.Wrap(controllerutil.SetOwnerReference(&svc, netpol, r.Scheme))
 }
 
-func (r *Reconciler) buildNetworkPolicy(ctx context.Context, ep effectivepolicy.ServiceEffectivePolicy) (*v1.NetworkPolicy, bool, error) {
+func (r *Reconciler) buildNetworkPolicies(ctx context.Context, ep effectivepolicy.ServiceEffectivePolicy) ([]v1.NetworkPolicy, bool, error) {
+	networkPolicies := make([]v1.NetworkPolicy, 0)
```

Comment:
(this applies to the next line)
the policyTypes slice is only relevant for the !ShouldCreateSeparatePolicies case, so better collect it in the `else` block rather than here.